spring-kafka/src/main/java/org/springframework/kafka/retrytopic/DeadLetterPublishingRecovererFactory.java

```diff
@@ -104,24 +113,32 @@ private int getAttempts(ConsumerRecord<?, ?> consumerRecord) {
 	}
 
 	private Headers addHeaders(ConsumerRecord<?, ?> consumerRecord, Exception e, int attempts) {
-		Headers headers = filterPreviousDltHeaders(consumerRecord.headers());
+		// TODO: Integrate better with DeadLetterPublisherRecoverer so that headers end up better
+		// TODO: Maybe change all timestamps to long instead of String (String is better to read but worse for performance)
```

Comment:
TODOs are useful, but creating a backlog issue for each is better.

This latter one would need to be done before 2.7 is released, or we'll have to handle both versions.

Reply:
Ok, I'll remove the todo's and write them on my notes so we can open the issues later. My main intention with those was letting you know of current shortcomings / decisions to be made.

That second TODO is actually outdated, I have changed all timestamps to long. I'll remove that.